test/exec.ts

```diff
@@ -1,59 +1,64 @@
-import { execFileSync } from 'child_process'
+import {
+  IConfiguration,
+  loadConfiguration,
+  runCucumber,
+} from '@cucumber/cucumber/api'
+import * as glob from 'glob'
 import { join } from 'path'
+import { PassThrough } from 'stream'
+import * as streamToString from 'stream-to-string'
+import { promisify } from 'util'
 
 import { ThemeStyles } from '../src/theme'
 
-const cmd = 'node_modules/.bin/cucumber-js'
-
-type RunOptionalOptions = {
-  '--name'?: string
-  '--tags'?: string[]
-}
-type RunOptions = {
-  colorsEnabled?: boolean
-  theme?: Partial<ThemeStyles>
+type FormatOptions = {
+  colorsEnabled: boolean
+  theme: Partial<ThemeStyles>
 }
-type FinalRunOptions = RunOptionalOptions & Required<RunOptions>
 
-export const run = (
+export const run = async (
   fileName: string,
-  options: RunOptions & RunOptionalOptions = {},
+  cucumberOptions: Partial<IConfiguration> = {},
+  formatOptions: Partial<FormatOptions> = {},
   throws = false
-): string => {
-  const { colorsEnabled, theme }: FinalRunOptions = {
-    colorsEnabled: false,
-    theme: {},
-    ...options,
-  }
-  const args = [
-    '--publish-quiet',
-    '--require',
-    join(__dirname, 'features'),
-    '--format',
-    join(__dirname, '..', 'src'),
-    '--format-options',
-    JSON.stringify({ colorsEnabled, theme }),
-  ]
-  if (options['--name']) args.push('--name', options['--name'])
-  if (options['--tags']) args.push('--tags', options['--tags'].join(','))
-
-  return exec(throws, ...args, join('test', 'features', fileName)).replace(
-    /\d+m\d+\.\d+s/g,
-    '0m00.000s'
-  )
-}
-
-const exec = (throws: boolean, ...args: string[]): string => {
-  if (process.env.LOG_CUCUMBER_RUN) console.log(`${cmd} ${args.join(' ')}`)
+): Promise<string> => {
+  // clear require cache for support code
```

Comment:
When we switch this package to ESM, we can just load the support code once in a top-level await and use that in every call.